validator/client/src/main/java/tech/pegasys/teku/validator/client/metrics/MetricRecordingValidatorApiChannel.java

```diff
@@ -0,0 +1,194 @@
+    sendAggregateRequestCounter =
+        metricsSystem.createCounter(
+            TekuMetricCategory.VALIDATOR,
+            PUBLISHED_AGGREGATE_COUNTER_NAME,
+            "Counter recording the number of signed aggregate attestations sent to the beacon node");
+    sendBlockRequestCounter =
+        metricsSystem.createCounter(
+            TekuMetricCategory.VALIDATOR,
+            PUBLISHED_BLOCK_COUNTER_NAME,
+            "Counter recording the number of signed blocks sent to the beacon node");
+  }
+
+  @Override
+  public SafeFuture<Optional<ForkInfo>> getForkInfo() {
+    return countRequest(delegate.getForkInfo(), forkInfoRequestCounter);
+  }
+
+  @Override
+  public SafeFuture<Optional<List<ValidatorDuties>>> getDuties(
+      final UnsignedLong epoch, final Collection<BLSPublicKey> publicKeys) {
+    return countRequest(delegate.getDuties(epoch, publicKeys), dutiesRequestCounter);
+  }
+
+  @Override
+  public SafeFuture<Optional<BeaconBlock>> createUnsignedBlock(
+      final UnsignedLong slot, final BLSSignature randaoReveal) {
+    return countRequest(
+        delegate.createUnsignedBlock(slot, randaoReveal), unsignedBlockRequestsCounter);
+  }
+
+  @Override
+  public SafeFuture<Optional<Attestation>> createUnsignedAttestation(
+      final UnsignedLong slot, final int committeeIndex) {
+    return countRequest(
+        delegate.createUnsignedAttestation(slot, committeeIndex),
+        unsignedAttestationRequestsCounter);
+  }
+
+  @Override
+  public SafeFuture<Optional<Attestation>> createAggregate(final AttestationData attestationData) {
+    return countRequest(delegate.createAggregate(attestationData), aggregateRequestsCounter);
+  }
+
+  @Override
+  public void subscribeToBeaconCommitteeForAggregation(
+      final int committeeIndex, final UnsignedLong aggregationSlot) {
+    subscribeAggregationRequestCounter.inc();
+    delegate.subscribeToBeaconCommitteeForAggregation(committeeIndex, aggregationSlot);
+  }
+
+  @Override
+  public void subscribeToPersistentSubnets(final Set<SubnetSubscription> subnetSubscriptions) {
+    subscribePersistentRequestCounter.inc();
+    delegate.subscribeToPersistentSubnets(subnetSubscriptions);
+  }
+
+  @Override
+  public void sendSignedAttestation(final Attestation attestation) {
+    sendAttestationRequestCounter.inc();
+    delegate.sendSignedAttestation(attestation);
+  }
+
+  @Override
+  public void sendAggregateAndProof(final SignedAggregateAndProof aggregateAndProof) {
+    sendAggregateRequestCounter.inc();
+    delegate.sendAggregateAndProof(aggregateAndProof);
+  }
+
+  @Override
+  public void sendSignedBlock(final SignedBeaconBlock block) {
+    sendBlockRequestCounter.inc();
+    delegate.sendSignedBlock(block);
+  }
+
+  private <T> SafeFuture<Optional<T>> countRequest(
+      final SafeFuture<Optional<T>> request, final BeaconChainRequestCounter counter) {
+    return request
+        .catchAndRethrow(__ -> counter.onError())
```

Comment:
Will catching the exceptions here affect the stack traces or how we're debugging validator API handler bugs?